backend/internal/handlers/utils.go

```diff
@@ -0,0 +1,81 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/golang/gddo/httputil/header"
+)
+
+type malformedRequest struct {
+	status int
+	msg    string
+}
+
+func (mr *malformedRequest) Error() string {
```

Comment:
nit: you don't need to use `malformedRequest` as a pointer everywhere

Reply:
I dug into this a bit and it looks like there's a good reason for using pointer receivers for custom error types

https://stackoverflow.com/a/50333850

Reply:
IMO `malformedRequest{status: 400, msg: "foo"} == malformedRequest{status: 400, msg: "foo"}` is a good property to have.
The linked post doesn't really explain why it's a bad thing, even in `io.EOF` case.

Reply:
Hmm yeah I suppose I hastily assumed that random guy on the internet had a good point because he wrote a lot of detail, but I also don't see a reason why and agree that the `io.EOF` case doesn't cut it. Will change.